packages/gateway/src/gateway-image-model.ts

```diff
@@ -64,6 +66,8 @@ export class GatewayImageModel implements ImageModelV3 {
           ...(size && { size }),
           ...(aspectRatio && { aspectRatio }),
           ...(seed && { seed }),
+          ...(files && { files }),
```

Comment:
if the type of the params for files or mask is uint8array we will send a verbose massive int array (which I believe will also not get deserialized into the right type server side). So, *if* it is a uint8array, then we need to do an encode/decode for this case. we can do it as a separate PR but please make sure to follow up with this work soon since it can be a common case and I know it will not work well.

See how we encode/decode uint8array file attachments for language models already as an example:

https://github.com/vercel/ai/blob/main/packages/gateway/src/gateway-language-model.ts#L188

and there is a decode side in gateway server to go with it.